src/tokens/Assets.tsx

```diff
@@ -0,0 +1,443 @@
+      <Animated.View
+        style={[styles.listHeaderContainer, animatedListHeaderStyles]}
+        onLayout={handleMeasureListHeaderHeight}
+      >
+        <View
+          style={[styles.nonStickyHeaderContainer]}
+          onLayout={handleMeasureNonStickyHeaderHeight}
+        >
+          <AssetsTokenBalance showInfo={displayPositions} />
+        </View>
+        <TabBar items={tabBarItems} selectedIndex={activeTab} onChange={handleChangeActiveView} />
+      </Animated.View>
+
+      <AnimatedSectionList
+        contentContainerStyle={{
+          paddingBottom: insets.bottom,
+          opacity: listHeaderHeight > 0 ? 1 : 0,
+        }}
+        // ensure header is above the scrollbar on ios overscroll
+        scrollIndicatorInsets={{ top: listHeaderHeight }}
+        // @ts-ignore can't get the SectionList to accept a union type :(
+        sections={sections}
+        renderItem={renderAssetItem}
+        renderSectionHeader={renderSectionHeader}
+        keyExtractor={keyExtractor}
+        onScroll={handleScroll}
+        scrollEventThrottle={16}
+        ListHeaderComponent={<View style={{ height: listHeaderHeight }} />}
+      />
+      {/* TODO(ACT-918): render collectibles */}
+      {showClaimRewards && (
+        <Animated.View
+          style={[
+            styles.footerContainer,
+            { paddingBottom: Math.max(insets.bottom, Spacing.Regular16) },
+            animatedFooterStyles,
+          ]}
+          onLayout={handleMeasureListFooterHeight}
+        >
+          <Button
+            type={BtnTypes.SECONDARY}
+            size={BtnSizes.FULL}
+            text={t('assets.claimRewards')}
+            onPress={() => {
+              ValoraAnalytics.track(AssetsEvents.tap_claim_rewards)
+              navigate(Screens.DappShortcutsRewards)
+            }}
+          />
+        </Animated.View>
+      )}
+    </>
+  )
+}
+
+AssetsScreen.navigationOptions = {
+  ...headerWithBackButton,
+}
+
+function TabBar({
+  items,
+  selectedIndex,
+  onChange,
+}: {
+  items: string[]
+  selectedIndex: number
+  onChange: (item: string, selectedIndex: number) => void
+}) {
+  const handleSelectOption = (item: string, index: number) => () => {
+    onChange(item, index)
+    vibrateInformative()
+  }
+
+  // On a smaller device, if there are more than two tabs, use smaller gaps
+  // between tabs
+  const gap =
+    items.length > 2 && variables.width < DEVICE_WIDTH_BREAKPOINT
+      ? Spacing.Smallest8
+      : Spacing.Regular16
+
+  return (
+    <View style={[styles.tabBarContainer, { gap }]} testID="Assets/TabBar">
+      {items.map((value, index) => (
+        <Touchable
+          testID="Assets/TabBarItem"
+          key={value}
+          onPress={handleSelectOption(value, index)}
+          style={styles.tabBarTouchable}
+        >
+          <Text
+            style={[index === selectedIndex ? styles.tabBarItemSelected : styles.tabBarItem]}
+            numberOfLines={1}
+            ellipsizeMode="tail"
```

Comment:
nit: "tail" is the default: [docs](https://reactnative.dev/docs/text#ellipsizemode).